### DISCUSSION TITLE
how send calendarId to event?

### DISCUSSION THREAD

**Discussion by abrahamluiss** [spatie/laravel-google-calendar#250] [Q&A]
I'm doing this
    $event = new Event;
    $event->name = 'name';
    $event->description = 'description ';
    $event->startDateTime = Carbon\Carbon::now();
    $event->endDateTime = Carbon\Carbon::now()->addHour();
    $event->calendarId = '5';
    $event->save();

now 

    $eventId = Event::get()->first()->id;
    $event = Event::find(eventId, '5');
    $event->delete();

answer
{ "error": { "errors": [ { "domain": "global", "reason": "notFound", "message": "Not Found" } ], "code": 404, "message": "Not Found" } }

**Comment by tomcodes** [comment]
Is this calendarId `5` here to illustrate or are you litterally trying to load this calendarId?

My calendarId looks nothing like this, it looks more like a complicated email address.

Here is where you will get your calendar_id:

![image](https://user-images.githubusercontent.com/2213846/137190584-f5eed724-6c40-4b46-8817-fa5f7c61710a.png)

**Comment by henriquek3** [comment]
@abrahamluiss I also have this question, the code says that it is possible to inform an attribute that is the calendar id

```PHP
public function __set($name, $value)
{
    $name = $this->getFieldName($name);

    if (in_array($name, ['start.date', 'end.date', 'start.dateTime', 'end.dateTime'])) {
        $this->setDateProperty($name, $value);

        return;
    }

    if ($name == 'source') {
        $this->setSourceProperty($value);

        return;
    }

    Arr::set($this->googleEvent, $name, $value);
}
```



[Event.php#L162](https://github.com/spatie/laravel-google-calendar/blob/713dbd7f57b6dda4d53e4437f89a72862f13e47f/src/Event.php#L162)


did you manage to solve it in any other way?

**Comment by filipnestak** [comment]
```php
$event = new Event();
$event->name = 'A new event';
$event->description = 'Event description';
$event->startDateTime = Carbon::now();
$event->endDateTime = Carbon::now()->addHour();

$calendar = GoogleCalendarFactory::createForCalendarId('calendarid@domain.com');
$calendar->insertEvent($event);
```

try this one, you can not set `calendarId` as attribute of `Event` object

**Comment by arborrow** [comment]
@ abrahamluiss If it helps, I use the uniqid function to create an event id which I can then search for using get. 

`$calendar_event = new Event;
$calendar_event->id = uniqid();`

then I save that unique id (here the retreat table's calendar_id field) into the table that the event is associated with and later when I want to get that event and update it I can use:

`            $calendar_event = Event::find($retreat->calendar_id);
`

Does that help answer your question?

When I have seen the error you are sharing, it is because the calendar_id provided is incorrect. So you want to distinguish between the calendar id and the event id.

p.s. - feel free to look at how I [implemented using the unique id](https://github.com/arborrow/montserrat/blob/master/app/Http/Controllers/RetreatController.php).
